functional_widget/lib/src/parameters.dart

```diff
@@ -63,28 +61,25 @@ Reference _parameterToReference(ParameterElement element) {
 }
 
 Reference _typeToReference(element_type.DartType type) {
-  if (type == null) {
-    return null;
-  }
   if (type is element_type.FunctionType) {
     // final functionTyped = type.element as FunctionTypedElement;
     final t = _functionTypedElementToFunctionType(type);
     return t.type;
   }
-  final displayName =
-      type.getDisplayString(withNullability: false /* TODO upgrade to true */);
-  return displayName != null ? refer(displayName) : null;
+  final displayName = type.getDisplayString(withNullability: true);
+  return refer(displayName);
 }
 
 FunctionType _functionTypedElementToFunctionType(
   element_type.FunctionType element,
 ) {
   return FunctionType((b) {
-    return b
+    b
       ..returnType = _typeToReference(element.returnType)
       // TODO set appropriate `nullabilitySuffix`
```

Comment:
Not sure what this is used for, so left it as is.

Reply:
Turns out nullabilitySuffix does not need to be set here, at least I think so - after adding some more tests.